### PR TITLE
Adding warning messages when NuGet task does not find the files

### PR DESCRIPTION
**Pull request by satbai**
Fixes issue #4905
Helps with issue #5086

### CODE REVIEW

**Comment by keithrob** [comment]
Please refactor based on our discussion.

File: Tasks/NuGetCommand/nugetpublisher.ts [hunk 1]
```diff
@@ -66,8 +67,13 @@ export async function run(nuGetPath: string): Promise<void> {
             }
         });
 
-        if (filesList && filesList.length < 1)
-        {
+        let searchPatternArray = searchPatternInput.split(";");
+        tl.debug(`Found ${filesList.length} files out of ${searchPatternArray.length}`);
```
Comment (pspill) [inline Tasks/NuGetCommand/nugetpublisher.ts]: count of search patterns is not useful. Maybe show # of files and display the actual patterns used?

File: Tasks/NuGetCommand/Common/VstsNuGetCommandHelper.ts [hunk 2]
```diff
@@ -0,0 +1,14 @@
+"use strict";
+
+export function logFileNotFoundWarning(tl, filesList, searchPatternArray): void {
```
Comment (pspill) [inline Tasks/NuGetCommand/Common/VstsNuGetCommandHelper.ts]: This fix should go in to NuGetCommand and DotNetCoreCLI, and therefore this common logging function should be in tasks/nuget-task-common

File: Tasks/Common/nuget-task-common/FilesListHelper.ts [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+        filesList.forEach(file => {
+            tl.debug(`--File: ${file}`);
+        });
+    } else {
+        searchPatternArray.forEach((pattern: string) => {
+            let findOptions: tl.FindOptions = <tl.FindOptions>{};
+            let matchOptions: tl.MatchOptions = <tl.MatchOptions>{};
+            let filesFound: string[] = [];
+
+            filesFound = currentTl.findMatch(undefined, pattern, findOptions, matchOptions);
```
Comment (satbai) [inline Tasks/Common/nuget-task-common/FilesListHelper.ts]: combine lines 26 and 24.

File: Tasks/Common/nuget-task-common/FilesListHelper.ts [hunk 4]
```diff
@@ -0,0 +1,45 @@
+"use strict";
+
+const nutil = require("nuget-task-common/Utility");
+import * as tl from "vsts-task-lib/task";
+
+export function createFoundFilesList(currentTl, searchPatternInput: string, ignoreLegacyFind?: boolean, basePathForLegacyFind?: string): string[] {
+    let filesList: string[] = [];
+    let searchPatternArray: string[] = nutil.getPatternsArrayFromInput(searchPatternInput);
+    let useLegacyFind = currentTl.getVariable("NuGet.UseLegacyFindFiles") === "true";
```
Comment (satbai) [inline Tasks/Common/nuget-task-common/FilesListHelper.ts]: Change this line into 
`let useLegacyFind = ignoreLegacyFind ? null : currentTl.getVariable("NuGet.UseLegacyFindFiles") === "true";`

File: Tasks/Common/nuget-task-common/FilesListHelper.ts [hunk 5]
```diff
@@ -0,0 +1,45 @@
+"use strict";
+
+const nutil = require("nuget-task-common/Utility");
+import * as tl from "vsts-task-lib/task";
+
+export function createFoundFilesList(currentTl, searchPatternInput: string, ignoreLegacyFind?: boolean, basePathForLegacyFind?: string): string[] {
```
Comment (satbai) [inline Tasks/Common/nuget-task-common/FilesListHelper.ts]: rename into currentTaskLib and add type any

**Comment by bryanmacfarlane** [comment]
take or close

**Comment by jerickmsft** [comment]
@keithrob What's the plan for this one?

**Comment by keithrob** [comment]
@jerickmsft, 
Still relevant.  Need some resources to roll off other features to complete this.

**Pull request closed without merging**